File: src/C#/CreationalPatterns/Builder/Laptops/Product/Laptop.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Laptops.Product
{
    public class Laptop
    {
        public string Model { get; set; }

        public string CPUSeries { get; set; }

        public string GPUModel { get; set; }

        public string RAMType { get; set; }

        public int RAMSize { get; set; }

        public string DisplayType { get; set; }

        public string SSDType { get; set; }

        public int SSDCapacity { get; set; }

        public List<string> Extras { get; set; } = new List<string>();

        public void ShowDetails()
        {
            StringBuilder stringBuilder = new StringBuilder();
            stringBuilder.AppendLine(new string('-', 40));
            stringBuilder.AppendLine($"Model: {Model}");
            stringBuilder.AppendLine($"CPU Model: {CPUSeries}");
            stringBuilder.AppendLine($"GPU Model: {GPUModel}");
            stringBuilder.AppendLine($"RAM Type: {RAMType}");
            stringBuilder.AppendLine($"RAM Size: {RAMSize} GB");
            stringBuilder.AppendLine($"Display Type: {DisplayType}");
            stringBuilder.AppendLine($"SSD Type: {SSDType}");
            stringBuilder.AppendLine($"SSD Capacity: {SSDCapacity} GB");
            stringBuilder.AppendLine($"Extras: ");
            stringBuilder.AppendLine(string.Join('\n', Extras.Select((e, i) => $"Extra #{i + 1}: {e}")));
            stringBuilder.AppendLine(new string('-', 40));
            Console.WriteLine(stringBuilder.ToString());
        }
    }
}

File: src/JavaScript/CreationalPatterns/Builder/Laptops/Product/laptop.js

class Laptop {
    constructor() {
        this.model = '';
        this.cpuSeries = '';
        this.gpuModel = '';
        this.ramType = '';
        this.ramSize = 0;
        this.displayType = '';
        this.ssdType = '';
        this.ssdCapacity = 0;
        this.extras = [];
    }

    showDetails() {
        const separator = '-'.repeat(40);
        const extrasString = this.extras.map((extra, i) => `Extra #${i + 1}: ${extra}`).join('\n');

        const detailsString = `${separator}
            Model: ${this.model}
            CPU Model: ${this.cpuSeries}
            GPU Model: ${this.gpuModel}
            RAM Type: ${this.ramType}
            RAM Size: ${this.ramSize} GB
            Display Type: ${this.displayType}
            SSD Type: ${this.ssdType}
            SSD Capacity: ${this.ssdCapacity} GB
            Extras:
            ${extrasString}
            ${separator}
        `;

        console.log(detailsString);
    }
}

module.exports = Laptop;
